File: ios/Samples/Swift/Liferay-iOS-Screens-Swift-Sample/AssetListViewController.swift
//
//  AssetListViewController.swift
//  Liferay-iOS-Screenlets-Swift-Sample
//
//  Created by jmWork on 11/09/14.
//  Copyright (c) 2014 Liferay. All rights reserved.
//

import UIKit


public class AssetListViewController: UIViewController, AssetListScreenletDelegate {

	@IBOutlet weak var screenlet: AssetListScreenlet?

    override public func viewDidLoad() {
        super.viewDidLoad()

        screenlet!.delegate = self
    }

	@IBAction func loadAction(sender: AnyObject) {
		SessionContext.createSession(
				username: "jose.navarro@liferay.com",
				password: "jm",
				userAttributes: ["userId": 10198])
		screenlet!.loadList()
	}

	public func onAssetListResponse(entries:[AssetListScreenlet.Entry]) {
		println("Loaded \(entries.count) entries")
		for e in entries {
			println("      \(e.title)")
		}
	}

	public func onAssetListError(error: NSError) {
	}

	public func onAssetSelected(entry: AssetListScreenlet.Entry) {
		println("selected \(entry.title)")
	}

}

File: xamarin/Samples/Showcase-iOS/ViewController/AssetListViewController.cs
using Foundation;
using LiferayScreens;
using System;
using UIKit;

namespace ShowcaseiOS.ViewController
{
    public partial class AssetListViewController : UIViewController, IAssetListScreenletDelegate
    {
        public AssetListViewController(IntPtr handle) : base(handle) { }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            //With classNameId
            //this.assetListScreenlet.ClassNameId = 20015;

            //With porletItemName
            this.assetListScreenlet.PortletItemName = "dynamic";

            this.assetListScreenlet.Delegate = this;
        }

        /* IAssetListScreenletDelegate */

        [Export("screenlet:onAssetListError:")]
        public virtual void OnAssetListError(AssetListScreenlet screenlet, NSError error)
        {
            Console.WriteLine($"Asset list error: {error.DebugDescription}");
        }

        [Export("screenlet:onAssetListResponse:")]
        public virtual void OnAssetListResponse(AssetListScreenlet screenlet, Asset[] assets)
        {
            Console.WriteLine($"Asset list response: {assets.Length} entries");
        }

        [Export("screenlet:onAssetSelected:")]
        public virtual void OnAssetSelected(AssetListScreenlet screenlet, Asset asset)
        {
            Console.WriteLine($"Asset selected: {asset.Title}");
        }
    }
}

